cpp/CMakeLists.txt

```diff
@@ -41,6 +41,7 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 # - User Options  ------------------------------------------------------------
 
 option(BUILD_TESTS "Build raft unit-tests" ON)
+option(BUILD_BENCH "Build raft C++ benchmark tests" ${BUILD_TESTS})
```

Comment:
We also have to get in the habit of updating build docs when we add/change options in raft. I admit I’ve not been great at doing this myself.

Reply:
Sorry, forgot about that!
Btw, you can see in the highlighted line, I've set the `BUILD_BENCH` be the same as `BUILD_TESTS` by default. Thus, raft's `--nogtest` flag effectively disables both. Would you like me (A) to keep this behavior and update the `BUILD.md` to reflect that, or (B) to introduce a new `--nobench` flag and add it everywhere in `ci` folder alongside `--nogtest`? I see the benefit of (B) is being more explicit, but (A) is less invasive and simple under the assumption that normally we need to disable/enable them both anyway.

(This also addresses your comment below).

Reply:
It looks like a bug in the cuml build.sh script. I’m assuming at some point support the `bench` option was removed inadvertently. 

Rather than using a `—bench` option, I’d prefer to invoke build.sh like `build.sh pyraft libraft bench`. Since the benchmarks are a top-level cmake project, this also follows the behavior that cuml *should* have (if it worked as documented). 

I would also prefer not to couple the gtests and the benchmarks options in cmake. Also, I would prefer `build.sh` build the benchmarks when no options are specified (just like how it builds the pyraft and libraft artifacts when no other r options are specified)

Reply:
Ok! I've added two targets (`tests` and `bench`) in `build.sh` in place of a single `--nogtest` option and changed the ci/conda scripts accordingly. Let's see if it passes :)